tools/generate_vsn.sh

```diff
@@ -1,9 +1,48 @@
 #!/bin/bash
+
+# generate_vsn.sh [type]
+# type = describe | tag | commits
+#
+# Certain 'type' values will produce following results:
+# - With "describe" output being "3.0.0-233-g984dd6df7"...
+# - "tag" will return "3.0.0"
+# - "commits" will return "233"
+# "describe" is the default value
+#
+# With git information missing, contents of VERSION file
+# will be returned for "describe" and "tag"; "0" will be returned
+# for "commits".
+
 DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
-GIT_VSN=`git describe --always --tags 2>/dev/null`
+VSN_TYPE=${1:-"describe"}
+
+echo_git_vsn() {
+    if [ "$VSN_TYPE" == "tag" ]; then
+        echo $LAST_TAG
+    elif [ "$VSN_TYPE" == "commits" ]; then
+        echo `git rev-list --count $LAST_TAG..HEAD 2>/dev/null`
```

Comment:
At first I thought "wait a sec, the $LAST_TAG var is undefined here". After some time I realised that we are inside a function and we use a global var. I know this is correct and we can live with this. On the other hand we could benefit from functional programming even in bash scripts.
This is just a comment, not a request for changes.

Reply:
Right. My basic skill in bash scripting instinctively told me to go for MVP. :D I'll improve it.